src/mysql-util/src/schemas.rs

```diff
@@ -99,7 +99,11 @@ where
         }
         SchemaRequest::Schemas(schemas) => {
             // Get all tables of type 'Base Table' in specified schemas
-            assert!(!schemas.is_empty());
+            if schemas.is_empty() {
```

Comment:
I would argue that we ought to return `Ok(vec![])` in these cases and not an error since we got asked to get the schema info of zero schemas (or zero tables) and that's exactly what we return. Then the logic in `events.rs` doesn't need to be special cased

Reply:
Ah I actually implemented it that way originally, but then thought that if we ever do get passed zero schemas or zero tables there is probably a logic error somewhere that is being hidden.

But I do like the general simplicity of it and am not too tied to that thought, so I'll switch it back!

Reply:
> if we ever do get passed zero schemas or zero tables there is probably a logic error somewhere that is being hidden

I thought that the issue description was saying the opposite, i.e that we entered this condition with a legitimate interleaving. Did I miss something?

Reply:
Nope -- you're right that it was a legitimate interleaving, as we had no tables we needed to verify. I guess I was future proofing against some other case, but that doesn't seem necessary.